src/components/ITSBStore/ITSBSearchHandler.jsx

```diff
@@ -0,0 +1,45 @@
+import { useEffect } from 'react';
+import { useRecoilState } from 'recoil';
+import { searchState, SearchStatus, useGraph } from '@peripleo/peripleo';
+
+/**
+ * The SearchHandler handles one thing: translating a Search
+ * into a list of results by querying the graph.
+ */
+export const ITSBSearchHandler = props => {
+
+  const [search, setSearchState] = useRecoilState(searchState);
+
+  const graph = useGraph();
+
+  useEffect(() => {
+    if (search.status === SearchStatus.PENDING) {
+      const authors = 
+        search.args.filters?.find(f => f.name === 'authors')?.values || [];
+
+      const dateRange = 
+        search.args.filters?.find(f => f.name === 'daterange')?.range;
+      
+      // TODO this could be optimized by filtering authors first!
+      const itineraries = graph.listItineraries(dateRange)
+        .filter(it => authors.includes(it.author));
+
+      setSearchState({
+        args: search.args,
+        status: SearchStatus.OK,
+        result: {
+          total: itineraries.length,
+          items: itineraries
+        }
+      });
+    }
+  }, [search]);
```

Comment:
I really like the elegance of having this all in one state object and updating when it changes, using the `PENDING`/`OK` statuses to avoid infinite re-renders.